python/social_bot/envs/grocery_ground.py

```diff
@@ -0,0 +1,336 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A simple enviroment for an agent play on a groceryground
+"""
+import os
+import time
+import logging
+import numpy as np
+import random
+import PIL
+import itertools
+
+import gym
+from gym import spaces
+from collections import OrderedDict
+
+import social_bot
+from social_bot import teacher
+from social_bot.teacher import TeacherAction
+from social_bot.teacher import DiscreteSequence
+from social_bot import teacher_tasks
+import social_bot.pygazebo as gazebo
+
+logger = logging.getLogger(__name__)
+
+
+class GroceryGroundGoalTask(teacher_tasks.GoalTask):
+    """
+    A simple teacher task to find a goal for env GroceryGround.
+    """
+
+    def __init__(self, reward_shaping=False, **kwargs):
+        """
+        Args:
+            eward_shaping (bool): if ture, use shaped reward accroding to distance rather than -1 and 1
+        """
+        super(GroceryGroundGoalTask, self).__init__(**kwargs)
+        self._reward_shaping = reward_shaping
+
+    def run(self, agent, world):
+        """
+        Start a teaching episode for this task.
+        Args:
+            agent (pygazebo.Agent): the learning agent 
+            world (pygazebo.World): the simulation world
+        """
+        agent_sentence = yield
+        agent.reset()
+        goal = world.get_agent(self._goal_name)
+        loc, dir = agent.get_pose()
+        loc = np.array(loc)
+        self._move_goal(goal, loc)
+        steps_since_last_reward = 0
+        while steps_since_last_reward < self._max_steps:
+            steps_since_last_reward += 1
+            loc, dir = agent.get_pose()
+            goal_loc, _ = goal.get_pose()
+            loc = np.array(loc)
+            goal_loc = np.array(goal_loc)
+            dist = np.linalg.norm(loc - goal_loc)
+            if dist < self._success_distance_thresh:
+                logger.debug("loc: " + str(loc) + " goal: " + str(goal_loc) +
+                             "dist: " + str(dist))
+                agent_sentence = yield TeacherAction(
+                    reward=10.0, sentence="Well done!", done=True)
+                steps_since_last_reward = 0
+                self._move_goal(goal, loc)
+            else:
+                if self._reward_shaping:
+                    agent_sentence = yield TeacherAction(
+                        reward=-0.1 * dist / self._random_range,
+                        sentence="Failed",
+                        done=False)
+                else:
+                    agent_sentence = yield TeacherAction()
+        logger.debug("loc: " + str(loc) + " goal: " + str(goal_loc) +
+                     "dist: " + str(dist))
+        yield TeacherAction(reward=-10.0, sentence="Failed", done=True)
+
+
+class GroceryGround(gym.Env):
+    """
+    The goal of this task is to train the agent to navigate to the objects given its
```

Comment:
It might mislead reader to think the name of the object is provided to the agent and agent need to understand the object name.

Change to the following:
"...navigate to a fixed type of object. The name of the object is provided in the constructor. In each episode, the location of the goal object is randomly chosen."

Reply:
done